students/IanMayther/Lesson09/Donor_Models.py

```diff
@@ -4,54 +4,151 @@
 
     def __str__(self):
         return str(self.name)
 
     def __repr__(self):
-        return "{}({})".format(self.name, self.donations)
+        return "{}".format(self.name)
+
+    @property
+    def initials(self):
+        don_name = ''
+        temp_list = re.findall('[A-Z][^A-Z]*', str(self.name))
+        for word in temp_list:
+            don_name += word[0]
+
+        return don_name
 
     def append(self,new_content):
+        '''Appends Donor donations'''
         if isinstance(new_content, (float, int)):
             self.donations.append(float(new_content))
         elif isinstance(new_content, list):
             for i in range(len(new_content)):
                 self.donations.append(new_content[i])
         return self.donations
 
+    def email(self):
+        '''Creates text for thank you email'''
+        body = f"""Greetings {self.name}\n
+        \n
+        Thank you so much for your generous contribution to our charity.\n
+        It is donors like you who make our work of building schools for ants' possible.\n
+        With your gift of ${math.fsum(self.donations)}, means (10) or (20) more schools can be built to help the ants learn to read.\n
+        \n
+        Sincerely,\n
+        Derek Zoolander\n
+        Founder and C.E.O. of Derek Zoolander Charity for Ants Who Can't Read Good (DZCAWCRG)\n"""
+        return body
+
+    def thank_you(self):
+        '''Creates Thank you for individual donors'''
+        body = "Thanks {} for your ${:.2f} in donations.".format(self.name, sum(self.donations))
+        return body
+
 
 class Donor_Collect(object):
     """
     Processes all the donors information doesn't work with donor functions
     """
 
-    # donors = []
-
     def __init__(self):
-        MS = Donor("Morgan Stanely")
-        '''
-        CV = Donor("Cornelius Vanderbilt")
-        JDR = Donor("John D. Rockefeller")
-        SG = Donor("Stephen Girard")
-        AC = Donor("Andrew Carnegie")
-        '''
-        self.donors = [MS]
+        self.donors = []
 
     def __str__(self):
         return "Collection of Donors: {}".format(str(self.donors))
 
     def __repr__(self):
-        return "{}".format(repr(self.donors))
\ No newline at end of file
+        return "{}".format(repr(self.donors))
+
+    def append(self,new_content):
+        '''Append Donors to collection'''
+        if isinstance(new_content, Donor):
+            self.donors.append(new_content)
+        else:
+            raise AttributeError("Only Class Donors can be append to .Donors list")
+        return self.donors
+
+    def calc_report(self):
+        ''''Creates table values for processing'''
+        new_dict = {}
+        for donor in self.donors:
+            new_dict[repr(donor)] = []
+            new_dict[repr(donor)].append(sum(donor.donations))
+            new_dict[repr(donor)].append(len(donor.donations))
+            new_dict[repr(donor)].append(sum(donor.donations)/len(donor.donations))
```

Comment:
the average, total donations and number of donations are good things to belong in the Donor class to create some abstraction.